dali/imgcodec/util/convert.cc

```diff
@@ -32,16 +32,14 @@ void Convert(SampleView<CPUBackend> out, TensorLayout out_layout, DALIImageType
   assert(in_shape.sample_dim() == out_shape.sample_dim());
   int ndim = out_shape.sample_dim();
   int spatial_ndim = ndim - 1;
```

Comment:
I think it would be good to add a comment about the assumption here and maybe use `spatial_ndim` through the code instead of `ndim - 1`.
When you explained it to me it is clear but without that it might not be obvious why.

Reply:
Thanks! I added a comment close to the call to the generic `Convert`, explaining that we assume channel-last layout, so that's why we pass `spatial_ndim` instead of `ndim`. I also extended the doc of the new Convert function to note that it's not just a wrapper choosing appropriate conversion function, but it also strips the channel dimension. I hope it's clearer now